Batched metafields and added logging

nick-groupe-ratte opened the pull request:
Resolves GR-10

nick-groupe-ratte committed:
Batched metafields and added logging

nick-groupe-ratte commented:
Hello @evanjamesjackson,
Please have a look

cameronfromtorq commented on src/Services/Stores/ShopifyStore.php:
```diff
@@ -361,63 +388,141 @@ public function commit(): Models\CommitResult
                 }
             } catch (Exception $e) {
                 $commitResults->addError(new LogRow("error during metafield setting in commit", $e->getMessage() . "\nFile: " . $e->getFile() . "\nLine: " . $e->getLine() . "\nTrace: " . $e->getTraceAsString()));
+                Logger::error("error during metafield setting in commit" . $e->getMessage() . "\nFile: " . $e->getFile() . "\nLine: " . $e->getLine() . "\nTrace: " . $e->getTraceAsString());
```
```suggestion

```

cameronfromtorq reviewed (changes requested):
Thank you for the PR! good find using the bulk-ish feature of metafield set. 

could you please move these changes out of your main branch into a feature branch. This way I can have two remotes in my development environment without branch name conflicts.

nick-groupe-ratte commented on src/Services/ExecutionService.php:
```diff
@@ -76,25 +113,41 @@ private function proccess($dataObject, &$rejects)
                     $this->storeInterface->updateProduct($dataObject);
                 }
-                foreach ($dataObject->getChildren([Concrete::OBJECT_TYPE_VARIANT], true) as $childVariant) {
+                if($variantCount > 0){
```
Yes you're right. I forgot to remove it

nick-groupe-ratte commented on src/Services/Stores/ShopifyStore.php:
```diff
@@ -39,17 +41,25 @@ class ShopifyStore extends BaseStore
     public function __construct(
         private ConfigurationRepository $configurationRepository,
         private ConfigurationService $configurationService,
+        ApplicationLogger $applicationLogger
```
Hello @cameronfromtorq  Should I do the same thing in the ExecutionService file ?

cameronfromtorq replied:
I just suggested this to keep it consistant within this file but ExecuitionService file isnt doing this.

nick-groupe-ratte commented on src/Services/Stores/ShopifyStore.php:
```diff
@@ -39,17 +41,25 @@ class ShopifyStore extends BaseStore
         $this->attributeService = new AttributesService();
-        $this->shopifyProductLinkingService = new ShopifyProductLinkingService($configurationRepository, $configurationService);
+        $this->shopifyProductLinkingService = new ShopifyProductLinkingService($configurationRepository, $configurationService, $applicationLogger);
+        $this->applicationLogger = $applicationLogger;
```
Hello @cameronfromtorq  Should I do the same thing in the ExecutionService file ?

nick-groupe-ratte committed:
Pull request changes

nick-groupe-ratte committed:
Logging

nick-groupe-ratte committed:
wip

Francismori7 committed:
Merge branch 'TorqIT:main' into main

nick-groupe-ratte committed:
WIP Stock

nick-groupe-ratte committed:
Product Linking

nick-groupe-ratte committed:
Merge branch 'main' into feat/gr-logging

nick-groupe-ratte committed:
Merge pull request #1 from grouperatte/feat/gr-logging

Feat/gr logging

nick-groupe-ratte committed:
WIP

nick-groupe-ratte committed:
Merge branch 'feat/gr-logging' of https://github.com/grouperatte/store-syndicator-bundle into feat/gr-logging

nick-groupe-ratte committed:
Dont update modification date if only inventory

nick-groupe-ratte committed:
Stock query

nick-groupe-ratte committed:
Update ShopifyStore.php

nick-groupe-ratte committed:
Update ShopifyQueryService.php

nick-groupe-ratte committed:
Images

nick-groupe-ratte committed:
Image upload

nick-groupe-ratte committed:
Update onPreSaveListener.php

nick-groupe-ratte committed:
Merge branch 'feat/gr-logging'

nick-groupe-ratte committed:
Ridestyler

nick-groupe-ratte committed:
Update ShopifyStore.php

nick-groupe-ratte committed:
Images and wheels

nick-groupe-ratte committed:
Cleanup

nick-groupe-ratte committed:
Stock update

LoganMacKinnon committed:
Merge remote-tracking branch 'TorqIt/feat/pimcore-11-upgrade' into feature/pim-11-update

LoganMacKinnon committed:
Fix composer lock

LoganMacKinnon committed:
Merge pull request #4 from grouperatte/feature/pim-11-update

Feature/pim-11-update

LoganMacKinnon committed:
Return type updates

LoganMacKinnon committed:
Type, extension updates, minor cleanup

LoganMacKinnon committed:
Merge pull request #5 from grouperatte/feature/upgrade-pimcore-11

feature/upgrade-pimcore-11

LoganMacKinnon committed:
Fix type issue

LoganMacKinnon committed:
Fix db connection type

LoganMacKinnon committed:
Fix unset array values

Francismori7 committed:
Update ShopifyAuthenticator.php

Force apiVersion to 2023-10 - will refactor later with @rayan-tsolefack because `createProduct` no longer supports variants

2023-10: https://shopify.dev/docs/api/admin-graphql/2023-10/mutations/productCreate
2024-07: https://shopify.dev/docs/api/admin-graphql/2024-07/mutations/productCreate
unstable: https://shopify.dev/docs/api/admin-graphql/unstable/mutations/productCreate

rayan-tsolefack committed:
fix(StoreSyndicator): The Shopify Export Error

rayan-tsolefack committed:
fix(StoreSyndicator): The Shopify Export Error

rayan-tsolefack committed:
fix(StoreSyndicator): The Shopify Export Error

rayan-tsolefack committed:
fix(StoreSyndicator): The Shopify Export Error

rayan-tsolefack committed:
fix(StoreSyndicator): The Shopify Export Error

rayan-tsolefack committed:
fix(StoreSyndicator): The Shopify Export Error

rayan-tsolefack committed:
fix(StoreSyndicator): The Shopify Export Error

arrandavidfraser committed:
Move createProduct processing to message handler for asyncronous processing.

arrandavidfraser committed:
Move createProduct processing to message handler for asyncronous processing.

arrandavidfraser committed:
Merge branch 'feat/asynchronous' of https://github.com/TorqIT/store-syndicator-bundle into feat/asynchronous

arrandavidfraser committed:
Corrected relative path reference to bundle MessageHandler folder

arrandavidfraser committed:
Added MessageBusInterface to abstract BaseStore so that StoreInterface can use it

arrandavidfraser committed:
Refactored: the entire ExecutionService::export loop is now in the message handler, making the processing asynchronous from admin UI while maintaining the processing in one loop/thread.

arrandavidfraser committed:
Added debug log on boot for StoreSydicationMessageHandler.

arrandavidfraser committed:
Remove unnecessary application log.

arrandavidfraser committed:
Refactor to comply with PHP8.2 deprecation of dynamic object properties; store variants array in separate array instead of in product objects.

LoganMacKinnon committed:
Update src/Controller/ExecutionController.php

LoganMacKinnon committed:
Update src/Controller/ExecutionController.php

LoganMacKinnon committed:
Update src/Controller/ExecutionController.php

LoganMacKinnon committed:
Merge pull request #6 from TorqIT/feat/asynchronous

Feat/asynchronous

arrandavidfraser committed:
Adds basic UI feedback to Manual Start button.

arrandavidfraser committed:
Merge pull request #7 from grouperatte/feat/asynchronous

Adds basic UI feedback to Manual Start button.

cameronfromtorq committed:
fix composer required version

cameronfromtorq committed:
add backend functionality to bulk publish products

cameronfromtorq committed:
query to get storefronts

cameronfromtorq committed:
link product to stores at end of job

arrandavidfraser committed:
To make this work, removed the check of local modification date against Shopify modification date.

arrandavidfraser committed:
Merge pull request #96 from TorqIT/feat/storefront-publishing

Feat/storefront publishing

arrandavidfraser committed:
Merge pull request #8 from grouperatte/feat/asynchronous

Adds working syndication for Shopify Sales Channels.

cameronfromtorq committed:
little bit of cleanup

cameronfromtorq committed:
uploading media and linking them to products

cameronfromtorq committed:
remove testing code and add some comments

arrandavidfraser committed:
Merge pull request #97 from TorqIT/feat/product-media

Feat/product media

arrandavidfraser committed:
Merge pull request #9 from grouperatte/feat/asynchronous

Adds new media handling via Shopify API

cameronfromtorq committed:
fix asset get path call

arrandavidfraser committed:
Merge pull request #10 from grouperatte/feat/asynchronous

fix asset get path call

cameronfromtorq committed:
use frontend path not full path

arrandavidfraser committed:
Merge pull request #11 from grouperatte/feat/asynchronous

use frontend path not full path

cameronfromtorq committed:
variant model restructure - fix prices & other fields

cameronfromtorq committed:
fix requires shipping

arrandavidfraser committed:
Correcting variable reference inside images array

arrandavidfraser committed:
Merge pull request #12 from grouperatte/feat/asynchronous

Correcting variable reference inside images array

cameronfromtorq committed:
remove variant creation from product creation flow and linking products outside of linking service

cameronfromtorq committed:
updating to use metafield query on create responses
also link variants on create

cameronfromtorq committed:
remove some old comments

cameronfromtorq committed:
Merge branch 'feat/asynchronous' into bugfix/variant-syncing

arrandavidfraser committed:
Merge pull request #98 from TorqIT/bugfix/variant-syncing

fixes after migrating to new api version & linking after creating prods & variants

arrandavidfraser committed:
Merge pull request #13 from grouperatte/feat/asynchronous

Rewrite product/variant API client for Shopify.

cameronfromtorq committed:
better logging

arrandavidfraser committed:
Merge pull request #99 from TorqIT/feat/better-logging

better logging

arrandavidfraser committed:
Merge pull request #14 from grouperatte/feat/asynchronous

Feat/asynchronous

cameronfromtorq committed:
fix create products logging

arrandavidfraser committed:
Merge pull request #15 from grouperatte/feat/asynchronous

fix create products logging

arrandavidfraser committed:
Asset image getFrontendFullPath() verified as using the CDN URL

arrandavidfraser committed:
Merge pull request #16 from grouperatte/feat/asynchronous

Asset image getFrontendFullPath() verified as using the CDN URL

arrandavidfraser committed:
Directly use CDN to handle public URL sent to Shopify for images.

arrandavidfraser committed:
Merge pull request #17 from grouperatte/feat/asynchronous

Directly use CDN to handle public URL sent to Shopify for images.

arrandavidfraser committed:
Add /pimcore-assets/ to the CDN prefix

arrandavidfraser committed:
Merge pull request #18 from grouperatte/feat/asynchronous

Add /pimcore-assets/ to the CDN prefix

arrandavidfraser committed:
A different approach to ensure /pimcore-assets/ path

arrandavidfraser committed:
Merge pull request #19 from grouperatte/feat/asynchronous

A different approach to ensure /pimcore-assets/ path

cameronfromtorq committed:
fix custom variant title functionality

cameronfromtorq committed:
fix requires shipping and inventory tracking fields

arrandavidfraser committed:
Merge pull request #100 from TorqIT/feat/small-variant-model-fixes

Feat/small variant model fixes

arrandavidfraser committed:
Merge pull request #20 from grouperatte/feat/asynchronous

Feat/asynchronous

arrandavidfraser committed:
Removing this condition means that all mapped fields are sent values to Shopify, even those values are empty.

arrandavidfraser committed:
Logging improvements.
Allow null/empty field mappings to be present in mutation data.  Handle null/empty checks in specific mapped fields.

Francismori7 committed:
Merge pull request #21 from grouperatte/feat/asynchronous

Feat/asynchronous

Francismori7 committed:
Update missing image

arrandavidfraser committed:
Handle images to Shopify in 2-step asynchronous process:
1 - upload image file
2a - attach image in Shopify to Product in Shopify
2b - if image wasn't ready, retry 2a

arrandavidfraser committed:
Remove URL used for local env testing

rayan-tsolefack committed:
Merging code from asynchronous

arrandavidfraser committed:
Corrected expected response structure - this response was being logged as error but actually worked on Shopify.

arrandavidfraser committed:
Improve handling of measurement weight of variants: avoid API errors by never sending weight or unit unless both are present and meaningful.

arrandavidfraser committed:
Attempt to convert abbreviated weight units to valid Shopify weight units

arrandavidfraser committed:
Prevent empty values for specific Shopify fields (compareAtPrice)

arrandavidfraser committed:
Merge pull request #23 from grouperatte/feat/asynchronous

Addresses 3 specific Shopify Errors

arrandavidfraser committed:
Additional check on empty fields that are Shopify data type "money"

arrandavidfraser committed:
Resolves error logged due to multidimensional array

arrandavidfraser committed:
Added context to debug log on failed image upload.

arrandavidfraser committed:
Merge pull request #24 from grouperatte/feat/asynchronous

More data type checks for Shopify money fields, improved log messages

arrandavidfraser committed:
Change query logs to debug
Add debug logs for upload image handler
Corrected property name for asset linking field
Changed handling of initial upload to explicity test for READY for instant linking

rayan-tsolefack committed:
Merge pull request #25 from grouperatte/fix/message-handling-file-status

More precise image upload handling

arrandavidfraser committed:
Fix: Pimcore will not log an Asset as a RelatedObject

arrandavidfraser committed:
Merge pull request #26 from grouperatte/fix/message-handling-file-status

Fix: Pimcore will not log an Asset as a RelatedObject

Francismori7 committed:
Remove pimcore-assets patching, there was an issue with CDN_URL on the supervisord container.

arrandavidfraser committed:
Corrected log entry for attaching image now/later.
Adding found status to log entry for error attaching READY image
Changing code to detect result of attach query.

arrandavidfraser committed:
Merge pull request #27 from grouperatte/fix/error-attaching-ready-image

Corrected log entry for attaching image now/later.

arrandavidfraser committed:
Typed testing on shopifyFileStatus results

arrandavidfraser committed:
Adding message file to attach handler logs

Francismori7 committed:
Merge pull request #28 from grouperatte/fix/attach-message-handler-logging

Adding message file to attach handler logs

arrandavidfraser committed:
More logs on attach image

arrandavidfraser committed:
Merge pull request #29 from grouperatte/fix/attach-message-handler-logging

More logs on attach image

arrandavidfraser committed:
One more log on pickup of attach image message

arrandavidfraser committed:
Change duplicate resolution mode to avoid recreating the same file

arrandavidfraser committed:
Remove Shopify File Id from Asset anytime it is saved

arrandavidfraser committed:
Separate out handling of new images vs images existing on Shopify.  Do not process images that are already linked.

arrandavidfraser committed:
Merge pull request #30 from grouperatte/fix/reuse-linked-images

Separate handling of images vs new images

arrandavidfraser committed:
Save the Shopify Product Id to the Image

arrandavidfraser committed:
Explicitly test for status error in file attach API response

arrandavidfraser committed:
Correct component name on Exception log

arrandavidfraser committed:
In memory mapping for storing Shopify product Ids to use on message bus.

arrandavidfraser committed:
Merge pull request #31 from grouperatte/fix/detect-need-to-retry-attach-image

Fix/detect need to retry attach image

arrandavidfraser committed:
Testing for $shopifyFileStatus == 1

arrandavidfraser committed:
Moved logging from message handler to ShopifyStore

arrandavidfraser committed:
Merge pull request #32 from grouperatte/fix/detect-attached-images

Fix/detect attached images

arrandavidfraser committed:
Additional logging for attach image handling.

arrandavidfraser committed:
Bugfix: code was expecting string result, not boolean

arrandavidfraser committed:
Improves asynchronous success on attaching images for new products; allow the message handler to query for store id if it wasn't found initially.

arrandavidfraser committed:
Merge pull request #33 from grouperatte/fix/image-attach-handling-create-product

Fix/image attach handling create product

arrandavidfraser committed:
Changed property names in event listener

arrandavidfraser committed:
Use StoreSyndicator Thumbnail image URLs instead of original files

arrandavidfraser committed:
When the thumbnail image is used, append .jpg to the filename if it is not already the extension.

rayan-tsolefack committed:
Merge pull request #34 from grouperatte/fix/check-image-filename-extension

Handles png assets converted to jpg by the thumbnail setting

Francismori7 committed:
patch to validate thumbnails are getting sent to shopify

LoganMacKinnon committed:
Add 1 minute delay to all AttachImageMessage dispatches

LoganMacKinnon committed:
Allow for delay to be determined by Pimcore website setting

Francismori7 committed:
Merge pull request #35 from grouperatte/feature/pim-1942-retry-delay

feature/pim 1942 retry delay

Francismori7 committed:
Revert "feature/pim 1942 retry delay"

Francismori7 committed:
Merge pull request #36 from grouperatte/revert-35-feature/pim-1942-retry-delay

Revert "feature/pim 1942 retry delay"

LoganMacKinnon committed:
Revert "Revert "feature/pim 1942 retry delay""

LoganMacKinnon committed:
Re-add deleted comment

arrandavidfraser committed:
Adds a test to exclude products from syncing if no variants have been loaded.

LoganMacKinnon committed:
Add attempt variable to shopifyattachimagemessage

LoganMacKinnon committed:
Add max retry attempt setting, add logic to not retry if reached

LoganMacKinnon committed:
Log final max attempt failure, remove properties from asset

LoganMacKinnon committed:
Only increase attempts when dispatched from handler

LoganMacKinnon committed:
Move logger after save

LoganMacKinnon committed:
Remove default attempt from wrapper

LoganMacKinnon committed:
Fix attempt argument defaults

LoganMacKinnon committed:
Set ShopifyAttachImageMessage default to 0

LoganMacKinnon committed:
Add increment comment

LoganMacKinnon committed:
Simplify retry logic

LoganMacKinnon committed:
Fix retry attempt check logic

LoganMacKinnon committed:
Final attempt logic fix after reading task

arrandavidfraser committed:
Merge pull request #37 from grouperatte/revert-36-revert-35-feature/pim-1942-retry-delay

Adds a delay between shopify attach API requests; configurable in Pimcore website settings.

LoganMacKinnon committed:
Add attempts to json encode function, improve variable naming

LoganMacKinnon committed:
Remove ShopifyFileStatus property after max retry attempts

LoganMacKinnon committed:
Fix off-by-one logic

LoganMacKinnon committed:
Simplify retry logic

arrandavidfraser committed:
remove extra json_encode()

arrandavidfraser committed:
Merge pull request #38 from grouperatte/feature/pim-1943-max-retries

Adds a retry limit to attach image API calls, configurable in Pimcore

arrandavidfraser committed:
Avoid deferred thumbnail generation

Francismori7 committed:
add logging for thumbnails and remove property deletion

Francismori7 committed:
update message handler to allow stock updates

Francismori7 committed:
revert changes to handler

rayan-tsolefack committed:
fix(datahub): issue with inventory item id update

rayan-tsolefack committed:
fix(datahub): issue with inventory item id update

rayan-tsolefack committed:
fix(datahub): issue with inventory item id update

Francismori7 committed:
bypass steps when in STOCKS export mode

rayan-tsolefack committed:
fix(datahub): Set prod store acces

rayan-tsolefack committed:
fix(Datamappers): Shopify Id update

rayan-tsolefack committed:
fix(commands): Push Tags to shopify

Francismori7 committed:
add seo fields to syndicator

arrandavidfraser committed:
Modify API query format, add money-format-awareness and custom metafields to bulk variant update query (copied from bulk variant create query)

Francismori7 committed:
Merge pull request #39 from grouperatte/fix/update-bulk-variant-query

Bulk Variant Update query

arrandavidfraser committed:
Removes pimcoreId from bulk variant update query

Francismori7 committed:
Merge pull request #40 from grouperatte/fix/remove-pimcoreId-from-bulk-variant-update

Removes pimcoreId from bulk variant update query

arrandavidfraser committed:
Adds a wait loop before adding products to stores in shopify export procedure; Adds CLI support to detect and delete a bulk operation if one becomes stuck.

Francismori7 committed:
Run stock exporter instead of main exporter when stocks config

cameronfromtorq committed:
remove double : on some pimcore object property names

arrandavidfraser committed:
Merge pull request #42 from grouperatte/bugfix/property-name-fix

remove double : on some pimcore object property names

cameronfromtorq committed:
also wait for "created" bulk mutation states and log if after running bulk mutations arent "completed" state

arrandavidfraser committed:
Added parentheses

arrandavidfraser committed:
Merge pull request #43 from grouperatte/feat/already-running-bulk-mutations

fix conflicting bulk mutation runs between steps

rayan-tsolefack committed:
my local version

rayan-tsolefack committed:
Removing debug logs
